src/renderer/renderScene.ts

```diff
@@ -205,32 +203,60 @@ const renderLinearPointHandles = (
   if (!centerPoint) {
     return;
   }
+  const { POINT_HANDLE_SIZE } = LinearElementEditor;
+
   points.forEach((point, idx) => {
     const isSelected =
       !!appState.editingLinearElement?.selectedPointsIndices?.includes(idx);
-    renderSingleLinearPoint(context, appState, renderConfig, point, isSelected);
+    const radius = appState.editingLinearElement
+      ? POINT_HANDLE_SIZE
+      : POINT_HANDLE_SIZE / 2;
+    renderSingleLinearPoint(
+      context,
+      appState,
+      renderConfig,
+      point,
+      radius,
+      isSelected,
+    );
   });
 
-  if (!appState.editingLinearElement && points.length < 3) {
+  if (points.length < 3) {
     if (appState.selectedLinearElement.midPointHovered) {
       const centerPoint = LinearElementEditor.getMidPoint(
         appState.selectedLinearElement,
       )!;
-      highlightPoint(centerPoint, context, appState, renderConfig);
-
-      renderSingleLinearPoint(
-        context,
-        appState,
-        renderConfig,
-        centerPoint,
-        false,
-      );
+      const radius = appState.editingLinearElement
+        ? POINT_HANDLE_SIZE
+        : POINT_HANDLE_SIZE / 2;
+      if (appState.editingLinearElement) {
+        renderSingleLinearPoint(
+          context,
+          appState,
+          renderConfig,
+          centerPoint,
+          radius,
+          false,
+        );
+        highlightPoint(centerPoint, context, renderConfig);
+      } else {
+        highlightPoint(centerPoint, context, renderConfig);
+        renderSingleLinearPoint(
+          context,
+          appState,
+          renderConfig,
+          centerPoint,
+          radius,
+          false,
+        );
+      }
```

Comment:
what's the difference between the branches? If either of those callbacks have side effects that work different based on order, then that's not great :D

Reply:
Its not about side effects but the final outcome
for editor -> the point radius is small but hover state is bigger and is on top of the point (as its full violet) so original point is not visible hence I am first rendering point and then hover state

<img width="152" alt="Screenshot 2022-08-18 at 7 31 26 PM" src="https://user-images.githubusercontent.com/11256141/185414163-b75be3cb-e92b-478f-8831-f58f7943e534.png">

outside editor -> The hover state has bigger radius but the original point is also visible as well hence  drawing the point after the hover state so its visible

<img width="85" alt="Screenshot 2022-08-18 at 7 31 02 PM" src="https://user-images.githubusercontent.com/11256141/185414062-a7bd4656-b8f8-4f73-9e34-700009c2167a.png">

Reply:
Mhm... I'd probably make the highlight part of `renderSingleLinearPoint()` instead. But barring that, let's add doc for this so it's clear.

Reply:
`highlight` I didn't make part of `renderSelectionElement` as highlight is only for active point and different for outside vs inside editor so kept the logic separate. Will add the doc.